### PR TITLE
Task/14295 wiring up SRS flow

### PR DESCRIPTION
**Pull request by naveeddotio**
## Description
Refactoring and wiring up SRS flow

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14277
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14295

### CODE REVIEW

**Review by CV113 — APPROVED** [review]
UA approved

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 1]
```diff
@@ -145,14 +151,14 @@ class ExposureSubmissionCoordinatorModel {
 		onSuccess: @escaping () -> Void,
 		onError: @escaping (ExposureSubmissionServiceError) -> Void
 	) {
-		guard let coronaTestType = coronaTestType else {
+		guard case let .registeredTest(coronaTestType) = submissionTestType, let testType = coronaTestType else {
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]: could you pls keep the name?
`let coronaTestType = coronaTestType`

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 2]
```diff
@@ -173,6 +179,29 @@ class ExposureSubmissionCoordinatorModel {
 		}
 	}
 
+	func submitSRSExposure(
+		isLoading: @escaping (Bool) -> Void,
+		onSuccess: @escaping () -> Void,
+		onError: @escaping (ExposureSubmissionServiceError) -> Void
+	) {
+		/*
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]: mark with a to.do comment?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 3]
```diff
@@ -342,11 +371,18 @@ class ExposureSubmissionCoordinatorModel {
...
-			coronaTestService.antigenTest.value?.isSubmissionConsentGiven = isSubmissionConsentGiven
+		switch submissionTestType {
+		case .registeredTest(let coronaTestType):
+			switch coronaTestType {
+			case .pcr:
+				coronaTestService.pcrTest.value?.isSubmissionConsentGiven = isSubmissionConsentGiven
+			case .antigen:
+				coronaTestService.antigenTest.value?.isSubmissionConsentGiven = isSubmissionConsentGiven
+			case .none:
+				fatalError("Cannot set submission consent, no corona test type is set")
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift:382]: Can .none ever happened?

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/SubmissionTestType.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+//
+
+import Foundation
+
+enum SubmissionTestType: Equatable {
+	case registeredTest(CoronaTestType?)
+	case srs(SRSSubmissionType)
+}
+
+enum SRSSubmissionType: Equatable {
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/SubmissionTestType.swift:14]: Would like to add a minor comment, why we wrap the `SAP_Internal_SubmissionPayload.SubmissionType` with that `SRSSubmissionType` enum type. (Cause of equatability)

**Review by flxschmidt — COMMENTED** [review]
Looks very good, already. Minor things as suggestions, or questions. :)

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift:382]: Probably not, added it because of the warning

**Review by flxschmidt — APPROVED** [review]

**Pull request merged**